### PR TITLE
[15.0][ADD] purchase_order_report_grouped_by_vendor: New module

### PR DESCRIPTION
**Pull request by pilarvargas-tecnativa**
cc @Tecnativa TT48783

@pedrobaeza @CarlosRoca13 please review

### CODE REVIEW

File: purchase_order_report_grouped_by_vendor/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+{
+    "name": "Purchase Order Report Grouped By Vendor",
+    "summary": "Purchase order report grouping orders by vendor",
+    "version": "15.0.1.0.0",
+    "category": "Purchase",
+    "website": "https://github.com/OCA/purchase-reporting",
+    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "depends": [
+        "purchase_requisition",
```
Comment (pedrobaeza) [inline purchase_order_report_grouped_by_vendor/__manifest__.py]: We should avoid this dependency for the generic module. Add this reference in custom. Put a hook method here if needed.

File: purchase_order_report_grouped_by_vendor/models/purchase_order.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+from odoo import models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def get_grouped_data(self):
+        grouped_data = defaultdict(lambda: {"partner_id": False, "order_ids": []})
+        for order in self:
+            partner_id = order.partner_id.id
```
Comment (pedrobaeza) [inline purchase_order_report_grouped_by_vendor/models/purchase_order.py]: Don't put .id, but store the recordset, and you avoid the latter browse.

File: purchase_order_report_grouped_by_vendor/views/report_purchase_order.xml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                </div>
+            </t>
+        </t>
+    </template>
+
+    <template id="report_purchase_order_grouped_by_vendor">
+        <t t-call="web.html_container">
+            <t
+                t-call="purchase_order_report_grouped_by_vendor.report_purchase_order_grouped_by_vendor_document"
+                t-lang="docs[:1].partner_id.lang"
```
Comment (pedrobaeza) [inline purchase_order_report_grouped_by_vendor/views/report_purchase_order.xml]: The loop should be done here for each different partner and pass the corresponding lang

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: purchase_order_report_grouped_by_vendor/__manifest__.py (hunk 1 above)
Comment (pilarvargas-tecnativa) [reply purchase_order_report_grouped_by_vendor/__manifest__.py]: The dependency was wrong, I should have changed it before pushing the module to the repository. 
All changes done, thanks

File: purchase_order_report_grouped_by_vendor/models/purchase_order.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+from odoo import models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def get_grouped_data(self):
+        grouped_data = defaultdict(lambda: {"partner": None, "order_ids": []})
+
```
Comment (pedrobaeza) [inline purchase_order_report_grouped_by_vendor/models/purchase_order.py]: Don't add empty lines inside a method.

File: purchase_order_report_grouped_by_vendor/views/report_purchase_order.xml [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+                                    </td>
+                                </tr>
+                            </tr>
+                        </tbody>
+                    </table>
+                    <!-- Insert page break after each vendor -->
+                    <div style="page-break-after: always;" />
+                </div>
+                </t>
+            </t>
```
Comment (pedrobaeza) [inline purchase_order_report_grouped_by_vendor/views/report_purchase_order.xml:132]: Weird indentation.

File: purchase_order_report_grouped_by_vendor/views/report_purchase_order.xml [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    </template>
+
+    <template id="report_purchase_order_grouped_by_vendor">
+            <t t-foreach="docs.get_grouped_data()" t-as="group">
```
Comment (pedrobaeza) [inline purchase_order_report_grouped_by_vendor/views/report_purchase_order.xml]: Weird indentation.

File: purchase_order_report_grouped_by_vendor/models/purchase_order.py (hunk 4 above)
Comment (pilarvargas-tecnativa) [reply purchase_order_report_grouped_by_vendor/models/purchase_order.py]: done!

**Review by pedrobaeza — APPROVED** [review]

**Review by carolinafernandez-tecnativa — APPROVED** [review]
Functional review LGTM :) Thanks!

**Pull request merged**